storage/remote/client.go

```diff
@@ -117,7 +118,12 @@ func (c *Client) Store(samples model.Samples) error {
 	defer httpResp.Body.Close()
 
 	if httpResp.StatusCode/100 != 2 {
-		err = fmt.Errorf("server returned HTTP status %s", httpResp.Status)
+		scanner := bufio.NewScanner(httpResp.Body)
+		line := ""
+		if scanner.Scan() {
+			line = scanner.Text()
```

Comment:
I'm assuming we're only doing one line of the returned body for it not to explode, in that case we should probably also limit the amount we read from that first line, as it can be arbitrary length.